src/main/java/com/javatemplate/api/auth/AuthController.java

```diff
@@ -21,12 +26,29 @@ public class AuthController {
 
     private final AuthenticationManager authenticationManager;
 
+    private final UserService userService;
+
     @PostMapping
     public JwtTokenResponseDTO login(final @RequestBody LoginDTO loginDTO) {
         final Authentication authentication = authenticationManager.authenticate(toAuthentication(loginDTO));
 
-        return JwtTokenResponseDTO.builder()
+        return JwtTokenResponseDTO
+                .builder()
                 .token(jwtTokenService.generateToken((JwtUserDetails) authentication.getPrincipal()))
                 .build();
     }
+
+    @PostMapping("/facebook")
+    public JwtTokenResponseDTO loginWithFacebook(final @RequestBody TokenRequestDTO tokenRequestDTO) {
+        final Facebook facebook = new FacebookTemplate(tokenRequestDTO.getAccessToken());
+        final String[] fields = {"email", "name"};
+
+        final User userLogin = facebook.fetchObject("me", User.class, fields);
+        final UserDetails userDetails = userService.loginWithFacebook(userLogin);
+
+        return JwtTokenResponseDTO
+                .builder()
```

Comment:
Do vấn đề ngữ nghĩa, thao tác `builder()` sẽ viết cùng dòng, tương tự cho `stream()`

```suggestion
        return JwtTokenResponseDTO.builder()
```

Reply:
Fixed